ui/packages/catalog-ui-search/src/main/webapp/component/filter/filter.view.js

```diff
@@ -159,23 +159,35 @@ the provided value."
   },
   onBeforeShow() {
     this.$el.toggleClass('is-sortable', this.options.isSortable || true)
-    this.filterAttribute.show(
-      DropdownView.createSimpleDropdown({
-        list: metacardDefinitions.sortedMetacardTypes
-          .filter(metacardType => !properties.isHidden(metacardType.id))
-          .filter(
-            metacardType => !metacardDefinitions.isHiddenType(metacardType.id)
-          )
-          .map(metacardType => ({
-            label: metacardType.alias || metacardType.id,
+    let filteredAttributeList = metacardDefinitions.sortedMetacardTypes
+      .filter(metacardType => !properties.isHidden(metacardType.id))
+      .filter(
+        metacardType => !metacardDefinitions.isHiddenType(metacardType.id)
+      )
+    filteredAttributeList = this.options.includedAttributes
+      ? filteredAttributeList.filter(metacardType =>
+          this.options.includedAttributes.includes(metacardType.id)
+        )
+      : filteredAttributeList
+
+    filteredAttributeList = filteredAttributeList.map(metacardType => ({
+      label: metacardType.alias || metacardType.id,
 
-            description: (properties.attributeDescriptions || {})[
-              metacardType.id
-            ],
+      description: (properties.attributeDescriptions || {})[metacardType.id],
 
-            value: metacardType.id,
-          })),
-        defaultSelection: [this.model.get('type') || 'anyText'],
+      value: metacardType.id,
+    }))
+
+    let defaultSelection = this.model.get('type') || 'anyText'
+    if (defaultSelection === 'anyText' && this.options.includedAttributes) {
```

Comment:
Should this only apply for the `anytext` attribute? If the model type isn't included in the `includedAttributes` should it also be changed